core/src/hlist.rs

```diff
@@ -928,6 +928,38 @@ where
     }
 }
 
+/// Implementation when target is reference and  the pluck target is in head
+impl<'a, T, Tail: ToRef<'a>> Plucker<&'a T, Here> for &'a HCons<T, Tail> {
+    type Remainder = <Tail as ToRef<'a>>::Output;
```

Comment:
While writing the test, I noticed that `Remainder` returns a value that is not an hlist. (example: `HCons<&String, &HCons<&i32, &HNil>>`)
I fixed it by adding `ToRef` so that it returns an hlist. (example: `HCons<&String, HCons<&i32, HNil>>`)
The same goes for `ByNameFieldPlucker`.